devenv/checks/dockerDesktop.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+import time
+
+from devenv.lib import proc
+from devenv.lib_check.types import checker
+from devenv.lib_check.types import fixer
+
+tags: set[str] = {"builtin"}
+name = "docker desktop shouldn't be running"
+
+
+def docker_desktop_is_running() -> bool:
+    procs = proc.run(("/bin/ps", "-Ac", "-o", "comm"), stdout=True)
+    return "Docker Desktop" in procs.split("\n")
+
+
+@checker
+def check() -> tuple[bool, str]:
+    if docker_desktop_is_running():
+        return (
+            False,
+            "Docker Desktop is running. We don't support it, and it conflicts with colima.",
+        )
+
+    return True, ""
+
+
+@fixer
+def fix() -> tuple[bool, str]:
+    # regular pkill won't stop the Docker Desktop UI,
+    # it'll just spin. the proper way to terminate it
+    # without SIGKILL is to use osascript.
+    print("Attempting to stop Docker Desktop.")
+    try:
+        proc.run(("osascript", "-e", 'quit app "Docker Desktop"'))
+    except RuntimeError as e:
+        return False, f"failed to quit Docker Desktop:\n{e}\n"
+
+    # osascript doesn't wait to make sure it finishes quitting
+    # so let's block for up to 5 secs
+    for i in range(10):
+        time.sleep(0.5)
+        if docker_desktop_is_running():
+            return True, ""
+
+    return True, ""
```

Comment:
wouldn't the last one be `False` if it failed to stop it after waiting?

Reply:
eh i guess it doesn't matter either way, it's just one less docker_desktop_is_running call (from the 2nd followup check being called) if this returns True

i'll change it to False to avoid confusion

Reply:
well acutally I do want to reuse the error message from check so returning True is preferred

Reply:
oh then isn't the loop backwards -- it's always going to return the first try if docker is still running

Reply:
Oh... you're right. I'm embarrassed lol, I guess my computer's just too fast